mindsdb/integrations/handlers/youtube_handler/youtube_tables.py

```diff
@@ -375,25 +380,27 @@ def get_videos_by_channel_id(self, channel_id):
         return self.get_videos_by_video_ids(video_ids)
 
     def get_videos_by_video_ids(self, video_ids):
-        resource = self.handler.connect().videos().list(part="statistics,snippet,contentDetails", id=",".join(video_ids)).execute()
-
         data = []
-        for item in resource["items"]:
-            data.append(
-                {
-                    "channel_id": item["snippet"]["channelId"],
-                    "channel_title": item["snippet"]["channelTitle"],
-                    "comment_count": item["statistics"]["commentCount"],
-                    "description": item["snippet"]["description"],
-                    "like_count": item["statistics"]["likeCount"],
-                    "publish_time": item["snippet"]["publishedAt"],
-                    "title": item["snippet"]["title"],
-                    "transcript": self.get_captions_by_video_id(item["id"]),
-                    "video_id": item["id"],
-                    "view_count": item["statistics"]["viewCount"],
-                    "duration_str": self.parse_duration(item["id"], item["contentDetails"]["duration"]),
-                }
-            )
+        # loop over 50 video ids at a time
+        # an invalid request error is caused otherwise
+        for i in range(0, len(video_ids), 50):
```

Comment:
Maybe we can just check if video_ids is a list

Reply:
Done.